services/static-webserver/client/source/class/osparc/component/editor/AnnotationNoteCreator.js

```diff
@@ -0,0 +1,159 @@
+/* ************************************************************************
+
+   osparc - the simcore frontend
+
+   https://osparc.io
+
+   Copyright:
+     2023 IT'IS Foundation, https://itis.swiss
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+
+   Authors:
+     * Odei Maiz (odeimaiz)
+
+************************************************************************ */
+
+qx.Class.define("osparc.component.editor.AnnotationNoteCreator", {
+  extend: qx.ui.core.Widget,
+
+  construct: function() {
+    this.base(arguments);
+
+    this._setLayout(new qx.ui.layout.VBox(10));
+
+    this.getChildControl("instructions");
+    this.getChildControl("select-destinatary");
+    this.getChildControl("note");
+    this.getChildControl("add");
+  },
+
+  properties: {
+    destinataryGid: {
+      check: "Integer",
+      init: null,
+      nullable: true,
+      event: "changeDestinataryGid"
+    },
+
+    note: {
+      check: "String",
+      init: "",
+      nullable: false,
+      event: "changeNote"
+    }
+  },
+
+  events: {
+    "addNote": "qx.event.type.Event",
+    "saveNote": "qx.event.type.Event",
+    "cancel": "qx.event.type.Event"
+  },
+
+  statics: {
+    popUpInWindow: function(noteEditor, newNote = true) {
+      const title = newNote ? qx.locale.Manager.tr("Add Note") : qx.locale.Manager.tr("Edit Note");
+      const win = osparc.ui.window.Window.popUpInWindow(noteEditor, title, 325, 256);
+      win.center();
+      win.open();
+      return win;
+    }
+  },
+
+  members: {
+    _createChildControlImpl: function(id) {
+      let control;
+      switch (id) {
+        case "instructions":
+          control = new qx.ui.basic.Label().set({
+            value: this.tr("Add a destinary and a notification will be sent. Please make sure the user has access to the ") + osparc.product.Utils.getStudyAlias() + ".",
```

Comment:
I would also make it shorter here.
```suggestion
            value: this.tr("Add a destinary to be notified. Please make sure the user has access to the ") + osparc.product.Utils.getStudyAlias() + ".",
```

Reply:
Would checking if the target user has access before displaying all the possible users be a good idea? I'm not sure what impact this could have.

Reply:
Yes, that was the initial idea, but the frontend can't know it if that user gets access via an organization.